index.js

```diff
@@ -144,6 +144,86 @@ function reset () {
   moduleAliasNames = []
 }
 
+//
+// Utility function
+//
+
+// Removes file name from the base and returns only dir.
+function getBaseDir (base) {
+  var parsed = nodePath.parse(base)
+  return parsed.ext ? parsed.dir : base
+}
+
+//
+// Functions to load configurations from a file
+//
+
+// Load configuration from any file
+function loadConfigFile (configPath) {
+  try {
+    var config = require(configPath)
+
+    // ES6 export default
+    if (config && config.default) {
+      return config.default
+    }
+
+    return config
+  } catch (e) {
+    // Do nothing
+  }
+}
+
+// Load configuration from the package.json file
+function loadPackageJSONFile (configPath) {
+  try {
+    var config = require(configPath)
+
+    if (config && config['module-alias']) {
+      return config['module-alias']
+    }
+
+    return {}
+  } catch (e) {
+    // Do nothing
+  }
+}
+
+function loadConfig (base, options) {
+  var configPath = ''
+  var config
+
+  // If a path has been provided, try loading the configuration using it
+  // It could be a simple JS, JSON or any file, or a package.json file
+  if (options.base) {
+    configPath = options.base
+
+    if (configPath.indexOf('package.json') > -1) {
+      config = loadPackageJSONFile(configPath)
+    } else {
+      config = loadConfigFile(configPath)
+    }
+  }
```

Comment:
This should already be covered by the following block in `init` : 
```javascript
  if (options.base) {
    candidatePackagePaths = [nodePath.resolve(options.base.replace(/\/package\.json$/, ''))]
  } else {
    ...
  }
```
Note here that we expect this to contain `package.json` ; but we should probably think about the case of `module-alias.config.js` as well.

Reply:
I don't think we need this at all since we are dealing with only `package.json`.
`[nodePath.resolve(options.base.replace(/\/package\.json$/, ''))]`

I think it equals to this now.
`candidatePackagePaths = [options.base]`

If the user has provided the path to a `package.json` file, it will be loaded.
If the user has provided the path to a `module-alias.config.js` file, it will be loaded.
If the user has provided the path to a dir, `module-alias.config.js` will load first, if failed, then `package.json`.

Reply:
Even better !